packages/gatsby-theme-api-docs/gatsby-config.js

```diff
@@ -21,6 +21,13 @@ module.exports = (themeOptions = {}) => {
           path: path.resolve('./src/api-specs'),
         },
       },
+      {
+        resolve: 'gatsby-source-filesystem',
+        options: {
+          name: 'type-locations',
+          path: path.resolve('./src/type-locations'),
```

Comment:
Why do we need a special folder for a system that is restricted to using one single file with a conventional file name?  It would be exposed already on GraphQL if it 's in the data folder and you could  still give it its own custom GraphQL facade if you want.